connectorx/src/sources/postgres/mod.rs

```diff
@@ -961,3 +966,771 @@ impl_produce!(
+                Some(s) => s,
+                None => panic!("get NULL for non-NULL column"),
+            },
+            SimpleQueryMessage::CommandComplete(c) => {
+                panic!("get command: {}", c);
+            }
+            _ => {
+                panic!("what?");
+            }
+        };
+        val
+    }
+}
+
+impl<'r, 'a> Produce<'r, Option<&'r str>> for PostgresSimpleSourceParser {
+    type Error = PostgresSourceError;
+
+    #[throws(PostgresSourceError)]
+    fn produce(&'r mut self) -> Option<&'r str> {
+        let (ridx, cidx) = self.next_loc()?;
+        let val = match &self.rows[ridx] {
+            SimpleQueryMessage::Row(row) => row.try_get(cidx)?,
+            SimpleQueryMessage::CommandComplete(c) => {
+                panic!("get command: {}", c);
+            }
+            _ => {
+                panic!("what?");
+            }
+        };
+        val
+    }
+}
+
+impl<'r> Produce<'r, Vec<u8>> for PostgresSimpleSourceParser {
+    type Error = PostgresSourceError;
+
+    #[throws(PostgresSourceError)]
+    fn produce(&'r mut self) -> Vec<u8> {
+        let (ridx, cidx) = self.next_loc()?;
+        let val = match &self.rows[ridx] {
+            SimpleQueryMessage::Row(row) => match row.try_get(cidx)? {
+                Some(s) => {
+
+                    let res: Vec<u8> = s.chars().map(|c| c as u8).collect::<Vec<_>>();
+                    res
+                },
+                None => panic!("get NULL for non-NULL column"),
+            },
+            SimpleQueryMessage::CommandComplete(c) => {
+                panic!("get command: {}", c);
+            }
+            _ => {
+                panic!("what?");
+            }
+        };
+        val
+    }
+}
+
+impl<'r, 'a> Produce<'r,Option<Vec<u8>>> for PostgresSimpleSourceParser {
+    type Error = PostgresSourceError;
+
+    #[throws(PostgresSourceError)]
+    fn produce(&'r mut self) -> Option<Vec<u8>> {
+        let (ridx, cidx) = self.next_loc()?;
+        let val = match &self.rows[ridx] {
+            SimpleQueryMessage::Row(row) => match row.try_get(cidx)?{
+                Some(s) => {
+                    let res: Vec<u8> = s.chars().map(|c| c as u8).collect::<Vec<_>>();
+                    Option::Some(res)
+                },
+                None => None
+            },
+            SimpleQueryMessage::CommandComplete(c) => {
+                panic!("get command: {}", c);
+            }
+            _ => {
+                panic!("what?");
+            }
+        };
+        val
+    }
+}
+
+fn rem_first_and_last(value: &str) -> &str {
+    let mut chars = value.chars();
+    chars.next();
+    chars.next_back();
+    chars.as_str()
+}
+
+fn parse_input_i16(input: &str) -> Result<Vec<i16>, std::num::ParseIntError> {
+    input
+        .split(",")
+        .map(|token| token.parse())
+        .collect()
+}
+
+impl<'r> Produce<'r, Vec<i16>> for PostgresSimpleSourceParser {
```

Comment:
This looks great! Is it possible to use a macro to handle all the vectors?

Reply:
Im having some issue imaging running a macro for these parsing functions

```
fn parse_input_i32(input: &str) -> Result<Vec<i32>, std::num::ParseIntError> {
    input.split(",").map(|token| token.parse()).collect()
}
```